Sources/TSCBasic/Process.swift

```diff
@@ -230,6 +230,7 @@ public final class Process {
     // process execution mutable state
     private var state: State = .idle
     private let stateLock = Lock()
+    private static let stateQueue = DispatchQueue(label: "ProcessStateQueue")
```

Comment:
we should consider allowing the user to specify the queue (with a default). also this is a serial one, correct?

Reply:
Global/concurrent queues are in general discouraged as a dispatch anti-pattern. They can create extra threads unnecessarily. I needed a queue to field the .notify() calls on. It's only a callback when the state machine calls the completion block. We could even do away with `stateLock` possibly.

Reply:
Added optional queue parameter in d67a7f0.